privacy-scaling-explorations/snark-artifacts
# feat: add `setup` script

Add an interactive setup bash script that will:
- clone the repo
- do some initial configuration related to sparse checkout (including defining a git alias to reuse the `--filter=blob:none`
  Unfortunately, from what I've read, even if sparse-checkout can avoid tracking files in the working dir, it won't prevent fetching everything remotely when one use `git fetch` or `git pull` (which defeats what we wanted: avoid downloading too much stuff)
- ask what to add/remove from sparse checkout

Later the same script can be run again (I wrapped it in an npm script for convenience) to skip the initial cloning and just ask to add or remove packages from sparse checkout.

Closes #21

## Test plan:
- in an already cloned repo:
  - checkout this branch
  - npm `setup`: should ask to add or remove package
  - press `a` or `r`
  - next you are asked to choose a package
  - press e, p or s
  - verify that the package was checked out or removed in your working dir
- test the initial cloning and setup:
  1. Delete your existing repo
  2.
      ```bash
      bash <(curl https://raw.githubusercontent.com/privacy-scaling-explorations/snark-artifacts/setup/scripts/bin/setup) fetch setup
      ```
  3. Follow the prompts. Check that what you wanted was checked out. 
  
  
### Comments
This became a bit complicated.
Was a fun exercise to learn a lot about git and bash. `sparse-checkout` is still a bit of an obscure option.
Not a perfect script. But I think we should not spend too much further time on this...Not sure the dx benefits are worth it.
